AndroidAnnotations/androidannotations-rest-spring/rest-spring/src/main/java/org/androidannotations/rest/spring/helper/RestSpringValidatorHelper.java

```diff
@@ -601,17 +600,12 @@ public void hasAnnotatedAllParameters(ExecutableElement element, ElementValidati
 		}
 
 		Set<String> urlVariableNames = restAnnotationHelper.extractUrlVariableNames(element);
-		if (urlVariableNames.size() != numberOfNotAnnotatedElementParameter(element) + numberOfPathAnnotatedParameter(element) + numberOfRequiresCookieInUrl(element)) {
-			validation.addError(element, "%s parameters must add annotations or define as @Path placeholders");
+		if (urlVariableNames.size() != numberOfPathAnnotatedParameter(element) + numberOfRequiresCookieInUrl(element)) {
+			validation.addError(element, "%s parameters must add annotations");
 		}
 
 		for (VariableElement variableElement : element.getParameters()) {
-			List<? extends AnnotationMirror> annotationMirrors = variableElement.getAnnotationMirrors();
-			if (annotationMirrors.size() == 0) {
-				if (!urlVariableNames.contains(variableElement.getSimpleName().toString())) {
-					validation.addError(element, "%s method parameters '" + variableElement + "' must be annotated");
-				}
-			} else if (!restAnnotationHelper.hasRequestParameterAnnotation(variableElement)) {
+			if (!restAnnotationHelper.hasRequestParameterAnnotation(variableElement)) {
```

Comment:
OK, the method name is not good. 

```java
	public boolean hasRequestParameterAnnotation(VariableElement variableElement) {
		return hasOneOfClassAnnotations(variableElement, Arrays.asList(Field.class, Part.class, Body.class, Path.class));
	}
```

How about `hasRestApiMethodParameterAnnotation`?